src/yaml-support/tkn-code-actions.ts

```diff
@@ -0,0 +1,147 @@
+        const startPos = document.positionAt(selectedElement.taskRef?.keyNode?.startPosition);
+        const endPos = document.positionAt(selectedElement.taskRef?.endPosition);
+        action.taskRefStartPosition = startPos;
+        action.taskRefEndPosition = endPos;
+        action.taskRefName = taskRefName;
+        action.taskKind = selectedElement.taskRef?.kind.value;
+        action.documentUri = document.uri;
+        result.push(action);
+      }
+    }
+
+    return result;
+  }
+  resolveCodeAction?(codeAction: TaskInlineAction): Thenable<vscode.CodeAction> {
+    return vscode.window.withProgress({location: vscode.ProgressLocation.Notification, cancellable: false, title: `Loading '${codeAction.taskRefName}' Task...` }, async (): Promise<vscode.CodeAction> => {
+      const uri = tektonFSUri(codeAction.taskKind === TektonYamlType.ClusterTask ? ContextType.CLUSTERTASK : ContextType.TASK , codeAction.taskRefName, 'yaml');
+      try {
+        const taskDoc = await tektonVfsProvider.loadTektonDocument(uri, false);
+        codeAction.edit = new vscode.WorkspaceEdit();
+        codeAction.edit.replace(codeAction.documentUri,
+          new vscode.Range(codeAction.taskRefStartPosition, codeAction.taskRefEndPosition), 
+          this.extractTaskDef(taskDoc, codeAction.taskRefStartPosition.character, codeAction.taskRefEndPosition.character));
+      } catch (err){
+        vscode.window.showErrorMessage('Cannot get Tekton Task definition: ' + err.toString());
+        console.error(err);
+      }
+      return codeAction;
+    });
+    
+  }
+
+  getProviderMetadata(): vscode.CodeActionProviderMetadata {
+    return {
+      providedCodeActionKinds: [vscode.CodeActionKind.RefactorInline.append('TektonTask')],
+    }
+  }
+
+  findTask(doc: TknDocument, pos: vscode.Position): PipelineTask | undefined {
+    const selectedElement = doc.findElementAt(pos);
+    if (!selectedElement) {
+      return undefined;
+    }
+    if (selectedElement.type === TknElementType.PIPELINE_TASK) {
+      return selectedElement as PipelineTask;
+    }
+
+    let parent = selectedElement.parent;
+    while (parent) {
+      if (parent.type === TknElementType.PIPELINE_TASK) {
+        return parent as PipelineTask;
+      }
+
+      parent = parent.parent;
+    }
+
+    return undefined;
+
+  }
+
+  private extractTaskDef(taskDoc: VirtualDocument, startPos: number, endPos): string {
+    const task: Task = jsYaml.safeLoad(taskDoc.getText()) as Task;
+
+    delete task.metadata.managedFields;
+    delete task.metadata.namespace;
+    delete task.metadata.resourceVersion;
+    delete task.metadata.selfLink;
+    delete task.metadata.uid;
+    delete task.metadata.generation;
+    delete task.metadata.creationTimestamp;
+    delete task.metadata.name;
+    delete task.metadata.ownerReferences;
+    delete task.metadata.annotations['kubectl.kubernetes.io/last-applied-configuration'];
```

Comment:
it fails here throw error `Cannot get Tekton Task definition: TypeError: Cannot convert undefined or null to object`